jwst/cube_build/ifu_cube.py

```diff
@@ -1204,17 +1262,374 @@ def map_detector_to_outputframe(self, this_par1, this_par2,
                 coord1, coord2 = coord.radec2std(self.crval1,
                                                  self.crval2,
                                                  ra_use, dec_use)
+
                 if self.weighting == 'miripsf':
                     alpha_det = alpha[good_data]
                     beta_det = beta[good_data]
             elif self.coord_system == 'alpha-beta':
                 coord1 = alpha[good_data]
                 coord2 = beta[good_data]
 
-        return coord1, coord2, wave, flux, rois_det, roiw_det, weight_det, \
+        return coord1, coord2, wave, flux, slice, rois_det, roiw_det, weight_det, \
             softrad_det, alpha_det, beta_det
 # ********************************************************************************
 
+    def map_fov_to_dqplane(self, this_par1, coord1, coord2, wave, roiw_ave, slice):
+        """ Set an initial DQ flag for the IFU cube based on FOV of input data
+
+        Map the FOV of channel (MIRI) or slice (NIRSPEC) to the DQ plane
+        and set an initial DQ flagging. The process is different for MIRI and NIRSpec.
+        In the MIRI case all the slices map roughly to the same FOV across the
+        wavelength range covered by the IFU. This is not the case for NIRSpec the
```

Comment:
Need punctuation between "NIRSpec" and "the". At least a semi-colon, if not a full stop (aka period).